src/pointcloud_to_laserscan_nodelet.cpp

```diff
@@ -139,40 +160,47 @@ namespace pointcloud_to_laserscan
   void PointCloudToLaserScanNodelet::cloudCb(const sensor_msgs::PointCloud2ConstPtr &cloud_msg)
   {
 
-    //build laserscan output
-    sensor_msgs::LaserScan output;
-    output.header = cloud_msg->header;
+    //build laserscan output for marking scan
+    sensor_msgs::LaserScan marking_output;
+    marking_output.header = cloud_msg->header;
     if (!target_frame_.empty())
     {
-      output.header.frame_id = target_frame_;
+      marking_output.header.frame_id = target_frame_;
     }
 
-    output.angle_min = angle_min_;
-    output.angle_max = angle_max_;
-    output.angle_increment = angle_increment_;
-    output.time_increment = 0.0;
-    output.scan_time = scan_time_;
-    output.range_min = range_min_;
-    output.range_max = range_max_;
+    marking_output.angle_min = angle_min_;
+    marking_output.angle_max = angle_max_;
+    marking_output.angle_increment = angle_increment_;
+    marking_output.time_increment = 0.0;
+    marking_output.scan_time = scan_time_;
+    marking_output.range_min = range_min_;
+    marking_output.range_max = range_max_;
+
+    // build laserscan output for clearing scan
+    sensor_msgs::LaserScan clearing_output(marking_output);
 
     //determine amount of rays to create
-    uint32_t ranges_size = std::ceil((output.angle_max - output.angle_min) / output.angle_increment);
+    uint32_t ranges_size = std::ceil((marking_output.angle_max - marking_output.angle_min) / marking_output.angle_increment);
+
 
     //determine if laserscan rays with no obstacle data will evaluate to infinity or max_range
     if (use_inf_)
     {
-      output.ranges.assign(ranges_size, std::numeric_limits<double>::infinity());
+      marking_output.ranges.assign(ranges_size, std::numeric_limits<double>::infinity());
     }
     else
     {
-      output.ranges.assign(ranges_size, output.range_max + 1.0);
+      marking_output.ranges.assign(ranges_size, marking_output.range_max + 1.0);
     }
 
+    //initialize with value to signal that no data was seen
```

Comment:
Would make the comments consistent with our style (undocumented) (// Sample comment)